com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -91,7 +91,7 @@ public bool HasScaleZ
             }
 
             public float PositionX, PositionY, PositionZ;
```

Comment:
Help me understand something, why are using separated float values for Pos, Rot and Scale, when we could implement a Vector3 Compressed wrapper class and all this ugly code could go away.

Reply:
we'd like to enable/disable individual components of a Vector3 for syncing through our `NetworkTransform.NetworkState` + present those options on the Inspector UI. in other words, we control the "compression" logic here.

Reply:
So, what would stop us from implementing that compression logic, + enable/disable individual components of a Vector3  + expose that to UI in a custom wrapper class?

Reply:
the main feature being implemented here is per axis part, bitwise flags & compression is an optional nice-to-have. what you're describing (`CompressedVector3` type) sounds like a feature on its own which would include a whole suite of network-optimized/compressed primitive types. considering the fact that Jeff just mentioned some potentials with snapshot delta-compression and Jaedyn planning to work on new FastBufferReader/Writer stuff makes me think that we should leave these wrappers for later for now at least. do you think that's fair?

Reply:
Okay, that's fair.